tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs

```diff
@@ -89,87 +89,116 @@ public static void Initialize()
+                    {
+                        Log.Error(ex, "Error sending CallTarget integration definitions to native library");
+                    }
+
+                    TelemetryFactory.Metrics.RecordDistributionSharedInitTime(MetricTags.InitializationComponent.CallTargetDefsPinvoke, sw.ElapsedMilliseconds);
+                    sw.Restart();
+
+                    InitializeTracer(sw);
+                }
+
+#if NETSTANDARD2_0 || NETCOREAPP3_1
+                try
+                {
+                    // On .NET Core 2.0-3.0 we see an occasional hang caused by OpenSSL being loaded
+                    // while the app is shutting down, which results in flaky tests due to the short-
+                    // lived nature of our apps. This appears to be a bug in the runtime (although
+                    // we haven't yet confirmed that). Calling the `ToUuid()` method uses an MD5
+                    // hash which calls into the native library, triggering the load.
+                    _ = string.Empty.ToUUID();
                 }
                 catch (Exception ex)
                 {
-                    Log.Error(ex, "Error sending CallTarget integration definitions to native library");
+                    Log.Error(ex, "Error triggering eager OpenSSL load");
                 }
+#endif
+                LifetimeManager.Instance.AddShutdownTask(RunShutdown);
 
-                TelemetryFactory.Metrics.RecordDistributionSharedInitTime(MetricTags.InitializationComponent.CallTargetDefsPinvoke, sw.ElapsedMilliseconds);
-                sw.Restart();
+                Log.Debug("Initialization finished.");
 
-                InitializeTracer(sw);
+                TelemetryFactory.Metrics.RecordDistributionSharedInitTime(MetricTags.InitializationComponent.Total, swTotal.ElapsedMilliseconds);
             }
-
-#if NETSTANDARD2_0 || NETCOREAPP3_1
-            try
+            catch (Exception ex) when (IsInSsi())
             {
-                // On .NET Core 2.0-3.0 we see an occasional hang caused by OpenSSL being loaded
-                // while the app is shutting down, which results in flaky tests due to the short-
-                // lived nature of our apps. This appears to be a bug in the runtime (although
-                // we haven't yet confirmed that). Calling the `ToUuid()` method uses an MD5
-                // hash which calls into the native library, triggering the load.
-                _ = string.Empty.ToUUID();
+                // if we're in SSI we want to be as safe as possible, so catching to avoid the possibility of a crash
+                try
+                {
+                    Log.Error(ex, "Error in Datadog.Trace.ClrProfiler.Managed.Loader.Startup.Startup(). Functionality may be impacted.");
+                }
+                catch
+                {
+                    // Swallowing any errors here, as something went _very_ wrong, even with logging
+                }
             }
```

Comment:
Is it in purpose that we bubble when we're not in Ssi?

Reply:
> Is it in purpose that we bubble when we're not in Ssi?

Yes, it is 🙂 https://github.com/DataDog/dd-trace-dotnet/pull/5838#discussion_r1700040669

The tl;dr; is that if we don't bubble, and we failed to log, you get a silent failure with no visible reason why, which could be very hard to debug for customers instrumenting in non-SSI.

In SSI, _not_ exposing the crash is more important than bubbling up, because we will be instrumenting far more stuff.

That said, if we _do_ log successfully, there's no reason to bubble up, so tweaked the code, cheers!🙂